packages/checkout-storefront/src/lib/auth/SaleorAuthClient.ts

```diff
@@ -67,7 +63,7 @@ export class SaleorAuthClient {
   };
 
   private handleRequestWithTokenRefresh: Fetch = async (input, init) => {
-    const refreshToken = this.storageHandler.getRefreshToken();
+    const refreshToken = this.storageHandler?.getRefreshToken();
 
     invariant(refreshToken, "Missing refresh token in token refresh handler");
```

Comment:
We allow `storageHandler` to be `undefined` but then this line will inevitably fail. We should either fix it or change it back to make `storage` required.

Reply:
Technically it shouldn't if this class flow works - this function runs only if you already have refreshToken, check `fetchWithAuth` method, but because refreshToken is not passed as a prop, but pulled from the class instance properties, according to typescript it could be null. So in case you're running this function without a refresh token it should fail because that means the whole flow is broken and it needs to be fixed